Neeva AI: Designs Part 2

Kokobe opened the pull request:
Adds Neeva AI designs in `CreateAccountScreen` and `PlansScreen`

## Checklists

### Pointers to help with review
* https://www.figma.com/file/6BWVIuiRpXepCbdZvlKK16/First-run---w-NeevaAI-messaging?node-id=3%3A2&t=dC65PlKV1TeJ75yP-0

### How was this tested?
- [x] I tested this manually on `debug`
- [ ] I tested this manually on `staging` or `release`
- [x] I added automated tests
- [ ] Existing automated tests cover some of my changes
- [ ] Existing automated tests cover all of my changes
- [x] I checked that `@Previews` still work in Android Studio


### Screenshots and screen recordings
<img width="1273" alt="Screen Shot 2023-02-23 at 9 53 54 AM" src="https://user-images.githubusercontent.com/20916043/220993534-91c91509-774a-4c2e-8413-0db33c874435.png">

<img width="1304" alt="Screen Shot 2023-02-23 at 10 11 42 AM" src="https://user-images.githubusercontent.com/20916043/220993661-4e4e7730-d4c1-4631-bd61-6f43ee4b4d73.png">


### Issues Referenced
* Fixes #1249

dan-alcantara-neeva commented on app/src/main/java/com/neeva/app/welcomeflow/PlansScreen.kt:
```diff
@@ -325,35 +313,29 @@ private fun SubscriptionInfo(
+private fun PlansBenefit(title: String, description: String?) {
+    Row {
+        Text(
+            text = "•",
```
I think we technically have this in the strings.xml file already.  It's "dot" or something?

dan-alcantara-neeva commented on app/src/main/java/com/neeva/app/welcomeflow/PlansScreen.kt:
```diff
@@ -325,35 +313,29 @@ private fun SubscriptionInfo(
-                openURL(Uri.parse(it.item))
-                return@AnnotatedSpannable
+        Column {
+            Spacer(modifier = Modifier.size(4.dp))
```
Dimensions.PADDING_TINY?

dan-alcantara-neeva reviewed (changes requested):
You'll need to figure out what to do about short screens:
screenshotTests/src/main/assets/golden/com.neeva.app.welcomeflow.WelcomeFlowScreenshotTests_createAccountScreen_OtherSignInOptions_Light_Preview.png

All this random text has pushed the actually useful "sign in" button off of the screen with no indication that it's there.

hwiatneeva commented:
- column 2: i have no solution of one design to cover the all cases. it needs a special layout in this size. 
- column 4: button text needs to be center-align, unless this language is more proper for left-aligned.

Kokobe commented:
> * column 2: i have no solution of one design to cover the all cases. it needs a special layout in this size.
> * column 4: button text needs to be center-align, unless this language is more proper for left-aligned.

Fixed! Updated the PR description
